### PR TITLE
hamid/82034/new-changes-of-the-country-selector

### PR DESCRIPTION
**Pull request by hamid-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 1]
```diff
@@ -16,6 +17,7 @@ const SignupWizard = ({ closeWizard }: TSignupWizardProps) => {
     const [is_cancel_wizard_dialog_active, setIsCancelWizardDialogActive] = React.useState(false);
     const [current_step_key, setCurrentStepKey] = React.useState<string>();
     const [is_country_selected, setIsCountrySelected] = React.useState(false);
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:19]: @hamid-deriv I think this state is redundant, You can just check if `selected_country` has a value or not at line `69`.

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx [hunk 2]
```diff
@@ -41,22 +43,13 @@ const CountrySelector = ({ onSelect, className }: TCountrySelectorProps) => {
...
-            WS.send();
-        }
-    }, [WS, residence_list]);
-
     return (
         <Formik initialValues={initial_form_values} validate={validateFields} onSubmit={submitCountry}>
             {({ errors, handleBlur, handleChange, setFieldValue, touched, values }) => (
                 <div className={className}>
-                    {!residence_list && is_loading && <Loading is_fullscreen={false} />}
                     {residence_list && residence_list.length && (
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx:53]: @hamid-deriv What's the point of this conditional rendering if we are not handling other cases? 🤔 
If `residence_list && residence_list.length` is not met then we render blank.

**Review by george-usynin-binary — APPROVED** [review]

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx (hunk 2 above)
Comment (hamid-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx:53]: When I use WS for fetching the list the select component throws an error when `resident_list` is not initialized. But I think I can remove it now, as the store initial value is `[]`. Thanks for tip :+1:

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 1 above)
Comment (hamid-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:19]: I'll check.

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx (hunk 2 above)
Comment (hamid-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx:53]: Sorted

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 1 above)
Comment (hamid-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx]: Sorted.

**Review by george-usynin-binary — APPROVED** [review]

**Pull request merged**